src/fuzz/protobufs/utils/spanner_emulator_ddl_statement_proto_to_string.cc

```diff
@@ -0,0 +1,188 @@
+std::string columnToPrimaryKey(Column column);
+std::string toString(Column::Orientation orientation);
+
+// Transforms any Emulator DDL statement into a syntactically valid string
+std::string toString(SpannerDDLStatement statement) {
+    using statementType = SpannerDDLStatement::DDLStatementCase;
+    switch (statement.DDLStatement_case()) {
+        case statementType::kCreateTable:
+            return toString(statement.createtable());
+        //TODO: add more cases here for additional APIs
+        default:
+            return "";
+    }
+}
+
+// generates a 'CREATE TABLE ...' statement
+std::string toString(CreateTable createTable) {
+    return absl::Substitute("CREATE TABLE $0 ([ $1,$2 ]) PRIMARY KEY ( [$3] )",
+                createTable.tablename(), 
+                toString(createTable.primarykeys()),
+                toString(createTable.nonprimarykeys()),
+                toPrimaryKeys(createTable.primarykeys()));
+}
+
+// converts a list of columns to a string format of {column1},{column2},...
+std::string toString(const google::protobuf::RepeatedPtrField<Column> columns) {
+    std::string columnsAsString = "";
+    for (int i = 0; i < columns.size(); i++) {
+        const Column& currCol = columns.at(i);
+        absl::StrAppend(&columnsAsString, toString(currCol));
+    }
+    // toString(currCol) will have an extra comma at the end so chop it off
+    columnsAsString = columnsAsString.substr(0, columnsAsString.size() - 1);
+    return columnsAsString;
+}
+
+// A column is { column_name data_type [ NOT NULL ] [ options_def ] }
+// where [ NOT NULL ] is optional and [ options_def ] is 
+// { OPTIONS ( allow_commit_timestamp = { true | null } ) }
+std::string toString(Column column) {
+    return absl::Substitute("{ $0 $1 $2 $3 },", 
+        column.columnname(), 
+        toString(column.columndatainfo()), 
+        isColumnNotNullToString(column.isnotnull()),
+        columnOptionsToString(column.allowcommittimestamp()));
+}
+
+// returns "data_type" or "ARRAY< data_type >" based on ColumnDataInfo
+std::string toString(ColumnDataInfo columnDataInfo) {
+    return columnDataInfo.isarray() ? 
+        absl::Substitute("ARRAY< $0 >", toString(columnDataInfo.scalartype(), 
+            columnDataInfo.length(), columnDataInfo.lengthtype())) :
+        toString(columnDataInfo.scalartype(), columnDataInfo.length(), 
+            columnDataInfo.lengthtype());
+}
+
+// returns the column's scalar type as a string
+std::string toString(ColumnDataInfo::ScalarType type, int length, 
+    ColumnDataInfo::LengthType lengthType) {
+    switch (type) {
+        case ColumnDataInfo::BOOL:
+            return "BOOL";
+        case ColumnDataInfo::INT64:
+            return "INT64";
+        case ColumnDataInfo::FLOAT64:
+            return "FLOAT64";
+        case ColumnDataInfo::STRING:
+            switch (lengthType) {
+                case ColumnDataInfo::BOUND:
+                    // ensures length is a positive number
```

Comment:
When can it be negative? Isn't there also STRING(MAX) to deal with?

Reply:
I spoke to Everett about this and my understanding is that it could be useful to allow unbounded lengths for string/bytes to find more bugs. While the documentation states the bounds, a user could still input the length incorrectly resulting in unexpected behavior.